File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/txpool/TxpoolStatusTest.java
package opensdk.sdk.apis.txpool;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.TxpoolStatusResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

@DisplayName("Txpool RPC Test")
public class TxpoolStatusTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC txpool_status")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        TxpoolStatusResponse response = sdk.txpool.status().send();
        response.getResult();
    }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/txpool/TxpoolStatus.test.js
const OpenSdk = require("@klaytn/web3rpc");
const { expect } = require("@jest/globals");
const { RPC } = require("../constant");

const sdk = new OpenSdk.TxpoolApi(new OpenSdk.ApiClient(RPC));

describe('txpool_status API', () => {
    test('should return txpool_status', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined()
            expect(typeof data.pending === 'number' || /^0x[0-9a-fA-F]+$/.test(data.pending)).toBe(true);
            done();
        };

        sdk.status({}, callbackOne);
    });
});

